classes.py

```
@@ -0,0 +1,109 @@
+В первом будут описаны все классы, во втором классы будут импортированы и
+использованы.
+"""
+
+
+@dataclass(eq=False, order=False)
+class Book:
+    pages: int
+    year: int
+    author: str
+    name: str
+    price: int
+    book_id: int = None
+
+    def __post_init__(self):
+        if not isinstance(self.pages, int):
+            raise NotIntException(self.pages)
+
+        if not isinstance(self.year, int):
+            raise NotIntException(self.year)
+
+        if not isinstance(self.author, str):
+            raise NotStrException(self.author)
+
+        if not isinstance(self.name, str):
+            raise NotStrException(self.name)
+
+        if not isinstance(self.price, int):
+            raise NotIntException(self.price)
+
+    def __str__(self):
+        return (f'{self.author}, {self.name}, {self.year} year, '
+                f'{self.pages} pages, {self.price} USD')
+
+    def __eq__(self, other):
+        return self.price == other.price
+
+    def __ne__(self, other):
+        return self.price != other.price
+
+    def __lt__(self, other):
+        return self.price < other.price
+
+    def __gt__(self, other):
+        return self.price > other.price
+
+    def __le__(self, other):
+        return self.price <= other.price
+
+    def __ge__(self, other):
+        return self.price >= other.price
+
+
+class Library:
+    def __init__(self):
+        self.library = []
+        self.next_id = 1
+
+    def add_book(self, book):
+        book.book_id = self.next_id
+        self.next_id += 1
+        self.library.append(book)
+
+    def get_book_info(self, authors):
+        if isinstance(authors, str):
+            author = [authors]
+        elif isinstance(authors, list):
+            author = authors
+        else:
+            raise TypeError("Поиск осуществляется по строке или списку")
+
+        if not self.library:
+            return 'Библиотека пуста'
+
+        if not authors:
+            return 'Подано пустое значение'
```

It's better to throw an error that you would handle outside the method

---

Fixed. Returning a handleable error.